Signal/src/contact/OWSContactsManager.m

```diff
@@ -240,170 +232,76 @@ - (void)updateSignalAccounts
+    self.signalAccounts = [signalAccounts copy];
+    [self.profileManager setContactRecipientIds:signalAccountMap.allKeys];
 }
 
-- (void)loadCachedDisplayNames
+// TODO dependency inject, avoid circular dependencies.
+- (OWSProfileManager *)profileManager
 {
-    // Read from database off the main thread.
-    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        NSMutableDictionary<NSString *, NSString *> *cachedAccountNameMap = [NSMutableDictionary new];
-        NSMutableDictionary<NSString *, NSString *> *cachedFirstNameMap = [NSMutableDictionary new];
-        NSMutableDictionary<NSString *, NSString *> *cachedLastNameMap = [NSMutableDictionary new];
-
-        [TSStorageManager.sharedManager.newDatabaseConnection readWithBlock:^(
-            YapDatabaseReadTransaction *_Nonnull transaction) {
-            [transaction
-                enumerateKeysAndObjectsInCollection:kTSStorageManager_AccountDisplayNames
-                                         usingBlock:^(
-                                             NSString *_Nonnull key, NSString *_Nonnull object, BOOL *_Nonnull stop) {
-                                             cachedAccountNameMap[key] = object;
-                                         }];
-            [transaction
-                enumerateKeysAndObjectsInCollection:kTSStorageManager_AccountFirstNames
-                                         usingBlock:^(
-                                             NSString *_Nonnull key, NSString *_Nonnull object, BOOL *_Nonnull stop) {
-                                             cachedFirstNameMap[key] = object;
-                                         }];
-            [transaction
-                enumerateKeysAndObjectsInCollection:kTSStorageManager_AccountLastNames
-                                         usingBlock:^(
-                                             NSString *_Nonnull key, NSString *_Nonnull object, BOOL *_Nonnull stop) {
-                                             cachedLastNameMap[key] = object;
-                                         }];
-        }];
-
-        if (self.cachedAccountNameMap || self.cachedFirstNameMap || self.cachedLastNameMap) {
-            // If these properties have already been populated from system contacts,
-            // don't overwrite.  In practice this should never happen.
-            OWSFail(@"%@ Unexpected cache state", self.logTag);
-            return;
-        }
-
-        self.cachedAccountNameMap = [cachedAccountNameMap copy];
-        self.cachedFirstNameMap = [cachedFirstNameMap copy];
-        self.cachedLastNameMap = [cachedLastNameMap copy];
-    });
+    return [OWSProfileManager sharedManager];
 }
 
 - (NSString *_Nullable)cachedDisplayNameForRecipientId:(NSString *)recipientId
 {
     OWSAssert(recipientId.length > 0);
 
-    return self.cachedAccountNameMap[recipientId];
+    SignalAccount *signalAccount = [self signalAccountForRecipientId:recipientId];
```

Comment:
`SignalAccount *_Nullable signalAccount` here and below...

Reply:
done